src/Decode.fs

```diff
@@ -154,239 +130,230 @@ module Decode =
+                jEl.TryGetUInt64()
+                |> function
+                    | true, uint64 -> Some uint64
+                    | _ -> None)
+            UInt64.TryParse
 
     let bigint : Decoder<bigint> =
         fun path token ->
             if Helpers.isNumber token then
-                Helpers.asInt token |> bigint |> Ok
+                Ok <| (Helpers.asInt token |> bigint)
             elif Helpers.isString token then
                 match bigint.TryParse (Helpers.asString token, NumberStyles.Any, CultureInfo.InvariantCulture) with
-                | true, x -> Ok x
-                | _ -> (path, BadPrimitive("a bigint", token)) |> Error
+                | true, bigint -> Ok bigint
+                | _ -> Error (path, BadPrimitive("a bigint", token))
             else
-                (path, BadPrimitive("a bigint", token)) |> Error
+                Error (path, BadPrimitive("a bigint", token))
 
     let bool : Decoder<bool> =
         fun path token ->
             if Helpers.isBool token then
-                Ok(Helpers.asBool token)
+                Ok <| Helpers.asBool token
             else
-                (path, BadPrimitive("a boolean", token)) |> Error
+                Error (path, BadPrimitive("a boolean", token))
 
     let float : Decoder<float> =
         fun path token ->
             if Helpers.isNumber token then
-                Helpers.asFloat token |> Ok
+                Ok <| Helpers.asFloat token
             else
                 (path, BadPrimitive("a float", token)) |> Error
 
     let float32 : Decoder<float32> =
         fun path token ->
             if Helpers.isNumber token then
-                Helpers.asFloat32 token |> Ok
+                Ok <| Helpers.asFloat32 token
             else
-                (path, BadPrimitive("a float", token)) |> Error
+                Error (path, BadPrimitive("a float", token))
 
     let decimal : Decoder<decimal> =
         fun path token ->
             if Helpers.isNumber token then
-                Helpers.asDecimal token |> Ok
+                Ok <| Helpers.asDecimal token
             elif Helpers.isString token then
-                match System.Decimal.TryParse (Helpers.asString token, NumberStyles.Any, CultureInfo.InvariantCulture) with
-                | true, x -> Ok x
-                | _ -> (path, BadPrimitive("a decimal", token)) |> Error
+                match Decimal.TryParse (Helpers.asString token, NumberStyles.Any, CultureInfo.InvariantCulture) with
+                | true, parsed -> Ok parsed
+                | _ -> Error (path, BadPrimitive("a decimal", token))
             else
-                (path, BadPrimitive("a decimal", token)) |> Error
+                Error (path, BadPrimitive("a decimal", token))
 
-    [<System.Obsolete("Please use datetimeUtc instead.")>]
-    let datetime : Decoder<System.DateTime> =
+    let datetimeUtc : Decoder<DateTime> =
         fun path token ->
-            if Helpers.isDate token then
```

Comment:
Similarly to numbers, `System.Text.Json` doesn't have a `ValueKind` for dates, so we have to use `string` and parse using `.TryGetDateTime()`